### DISCUSSION TITLE
App compilation not failing in dev mode for eslint errors

### DISCUSSION THREAD

**Discussion by himanshulyric** [vercel/next.js#69225] [Help]
### Summary

I have a next js monorepo using turbo. My app has eslint issues but in local dev environment (next dev) the compilation is not failing for the eslint issues. Is there any way the compilation of next js app  in dev environment fails like in CRA (Create React App)

**Comment by icyJoseph** [comment]
Could you run `next lint`  before running `next dev`, like: `next lint && next dev`, and assign that to a script in your package json?

Though that won't catch errors afterwards. In general, even TS errors don't cause failure, because, IIRC, it is perceived as getting in the way of getting stuff done.

**Reply by himanshulyric** [reply]
Let me give a little background. We have a nextjs monorepo using turbo js. i am able to run the lint if I run the command explicitly. What i am looking for is the dev server failing if there are linting errors, much like CRA (Create React App) where the developer is able to see the linting errors while developing a feature, instead of seeing those errors in  say pre-commit hook or some github actions. Any help will be much appreciated

**Reply by icyJoseph** [reply]
Yeah, I know what you want and I'm saying that, at least as far as typescript goes, these kind of checks during dev mode, were considered to be detrimental to the dev experience. You would NOT be able to just try out stuff, because the linter or type checker would bring your app down. Losing state and other bits. Just because CRA did it, doesn't mean it was a good idea.

- https://github.com/vercel/next.js/issues/14997

I guess, you could do something similar to what people do to enable typescript, on dev mode, by modifying your next config's, we pack configuration, but I'm not sure about that. Isn't it easier to perhaps, fix lint errors on save? I know it can have downsides with merge conflicts... But better than nothing.

**Reply by himanshulyric** [reply]
Devs often do miss to add the appropriate `useMemo/useCallBack` dependencies. If the devs are informed at a very later stage of development say in pre-commit hook or in github action, it becomes very inconvenient and also impacts dev testing. Anyways i understand that nextjs does not have this feature out of the box and we need to have a workaround solution. Thanks for this discussion @icyJoseph . Much appreciated.

**Reply by icyJoseph** [reply]
> Devs often do miss to add the appropriate `useMemo/useCallBack` dependencies. 

I can't recall any project, nor, text editor combination, where a properly configured linter wouldn't show errors right as you cause them. In the text editor, and highlight where the errors are. And even give the option to fix the issues.

If that's not the case for you, perhaps your project is somewhat misconfigured? Look into some Next.js starter repositories, and see what config's might be missing.

> We recommend using an appropriate integration to view warnings and errors directly in your code editor during development. [From the Next.js docs](https://nextjs.org/docs/app/building-your-application/configuring/eslint)

That recommendation links here, https://eslint.org/docs/latest/use/integrations#editors. Give it a read.
